docs/features/readonly-variables-and-parameters.md

```diff
@@ -0,0 +1,61 @@
+### High Level Language Proposal:  
+Allow 'readonly' to be used on local variables and on parameters. 
+
+'readonly' on local variables and 'parameters' will work similarly as it does for 'fields' (with important differences listed in the next paragraph).  Namely, 'readonly' local variables and parameters cannot be written to in any way (i.e. through assignment, use of ```++/+=```, passing as a 'ref/out', etc.). 
+
+Important differences:
+unlike 'readonly' fields, 'readonly' locals must have an initializer.
+unlike 'readonly' fields (which can be written to multiple times in constructors), 'readonly' locals can only have the initializer and can never be written to again.
```

Comment:
From talking to @gafter this is fixable (if we implement 'definite-unassignment').  I'll bring this issue to the LDM and see what we think should be done.  Right now i can think of three options:

1. Just allow the above.  Tell people that 'readonly' prevents writing to the local at any place but the initializer.  But there's nothing stopping the initializer from running multiple times.  Kinda wonky, but free to do.
2. Disallow readonly variables in labeled sections of code (with a message saying precisely why i.e. "because the variable may get written multiple times").  We would also need to do this for switches (though we could potentially only care about case-blocks that  We might only have to do it though if someone actually used goto in the switch.  Small-ish cost.  
3. Implement 'definite unassignment'.  Would likely get this completely right (as well as allowing the user to not have to initialize the local at the declaration point).  Would be very expensive according to Neal though.